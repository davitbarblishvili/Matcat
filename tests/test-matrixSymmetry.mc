// Author: Davit
func main() int {
    int test; 
    matrix a = [[1,2,3],[2,4,5],[3,5,6]];
    matrix b; 

    test = check_symmetry(a); 
    if(test == 1){
        printStr("The matrix is symmetric");
        printStr("Here is the original matrix:"); 
        printm(a); 
        printStr("");
        printStr("Here is the transposed matrix:"); 
        b = transpose(a);
        printm(b);
        
    }else{
        printStr("The matrix is not symmetric");
    }
}