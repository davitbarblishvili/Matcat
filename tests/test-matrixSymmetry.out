The matrix is symmetric
Here is the original matrix:
(
[1.00 2.00 3.00 ]
[2.00 4.00 5.00 ]
[3.00 5.00 6.00 ]
)

Here is the transposed matrix:
(
[1.00 2.00 3.00 ]
[2.00 4.00 5.00 ]
[3.00 5.00 6.00 ]
)